consumer/communication.go

```diff
@@ -244,28 +283,44 @@ func UEContextTransferRequest(
 }
 
 // This operation is called "RegistrationCompleteNotify" at TS 23.502
-func RegistrationStatusUpdate(ue *amf_context.AmfUe, request models.UeRegStatusUpdateReqData) (
-	regStatusTransferComplete bool, problemDetails *models.ProblemDetails, err error,
+func RegistrationStatusUpdate(ctx context.Context, ue *amf_context.AmfUe, request models.UeRegStatusUpdateReqData) (
+	bool, *models.ProblemDetails, error,
 ) {
+	ctx, span := tracer.Start(ctx, "HTTP POST amf/ue-contexts/{ueContextId}/transfer-update")
+	defer span.End()
+
+	span.SetAttributes(
+		attribute.String("http.method", "POST"),
+		attribute.String("nf.target", "amf"),
+		attribute.String("net.peer.name", ue.TargetAmfUri),
+		attribute.String("ue.supi", ue.Supi),
+		attribute.String("ue.plmn.id", ue.PlmnId.Mcc+ue.PlmnId.Mnc),
+	)
+
 	configuration := Namf_Communication.NewConfiguration()
 	configuration.SetBasePath(ue.TargetAmfUri)
 	client := Namf_Communication.NewAPIClient(configuration)
 
-	ctx, cancel := context.WithTimeout(context.TODO(), 30*time.Second)
+	ctx, cancel := context.WithTimeout(ctx, 30*time.Second)
 	defer cancel()
+
+	var problemDetails *models.ProblemDetails
+	var err error
+	var regStatusTransferComplete bool
```

Comment:
Same as my previous comment. I would suggest reverting this change (definition of the return variables)
```go
func RegistrationStatusUpdate(ctx context.Context, ue *amf_context.AmfUe, request models.UeRegStatusUpdateReqData) (
	regStatusTransferComplete bool, problemDetails *models.ProblemDetails, err error,
) {
```
```suggestion
func RegistrationStatusUpdate(ctx context.Context, ue *amf_context.AmfUe, request models.UeRegStatusUpdateReqData) (
	regStatusTransferComplete bool, problemDetails *models.ProblemDetails, err error,
) {
	ctx, span := tracer.Start(ctx, "HTTP POST amf/ue-contexts/{ueContextId}/transfer-update")
	defer span.End()

	span.SetAttributes(
		attribute.String("http.method", "POST"),
		attribute.String("nf.target", "amf"),
		attribute.String("net.peer.name", ue.TargetAmfUri),
		attribute.String("ue.supi", ue.Supi),
		attribute.String("ue.plmn.id", ue.PlmnId.Mcc+ue.PlmnId.Mnc),
	)

	configuration := Namf_Communication.NewConfiguration()
	configuration.SetBasePath(ue.TargetAmfUri)
	client := Namf_Communication.NewAPIClient(configuration)

	ctx, cancel := context.WithTimeout(ctx, 30*time.Second)
	defer cancel()

```

Reply:
Done